components/builder-api/src/server/resources/channels.rs

```diff
@@ -14,371 +14,326 @@
 
 use actix_web::http::{self, StatusCode};
 use actix_web::FromRequest;
-use actix_web::{App, HttpRequest, HttpResponse, Json, Path};
+use actix_web::{App, HttpRequest, HttpResponse, Json, Path, Query};
 use protocol::originsrv::*;
+use serde_json;
 
 use hab_core::package::ident;
+use hab_net::{ErrCode, NetError, NetOk};
 
 use server::error::{Error, Result};
 use server::framework::headers;
 use server::framework::middleware::route_message;
 use server::helpers;
 use server::AppState;
 
+#[derive(Debug, Default, Clone, Deserialize)]
+#[serde(default)]
+pub struct SandboxBool {
+    is_set: bool,
+}
+
 pub struct Channels;
 
 impl Channels {
     //
     // Internal - these functions should return Result<..>
     //
+    fn do_promote_package(
+        req: &HttpRequest<AppState>,
+        origin: String,
+        channel: String,
+    ) -> Result<()> {
+        let ident = OriginPackageIdent::new();
+
+        helpers::check_origin_access(req, &origin)?;
+
+        let mut origin_get = OriginGet::new();
+        origin_get.set_name(ident.get_origin().to_string());
+
+        let origin_id = route_message::<OriginGet, Origin>(req, &origin_get)?;
+
+        let mut channel_req = OriginChannelGet::new();
+        channel_req.set_origin_name(ident.get_origin().to_string());
+        channel_req.set_name(channel.to_string());
+
+        let origin_channel =
+            match route_message::<OriginChannelGet, OriginPackage>(req, &channel_req) {
+                Ok(c) => c,
+                Err(e) => {
+                    error!("Error retrieving channel: {:?}", &e);
```

Comment:
minor - switch error! to warn! .. maybe even debug! if these can happen under 'normal' operation